### PR TITLE
🐛Fixed asset redirects

### LINKED ISSUE #9445: Redirect image requests
If you have setup your blog with https and you request an image with http (e.g.  `http://myblog.com/content/images/1.jpg`, the image request doesn't get redirected to https.

This is because the redirect unit is defined [to late](https://github.com/TryGhost/Ghost/blob/1.20.3/core/server/web/site/app.js#L110). There are some comments in the code base, that url redirects have to happen after assets loading. This needs a little more investigation to ensure we are not breaking things.

Also: core assets should be redirected as well. All asset requests should use the correct protocol configured.

- [ ] reorder logic (ensure we don't prettify image requests)
- [ ] add tests

### PR DESCRIPTION
**Pull request by kirrg001**
closes #9445

- re-use and extend our url-redirect middleware
- add proper integration tests for our express site app (no db interaction, component testing required for such important use cases)
  - i added some more general tests
- tested in the browser
- redirects all asset requests if https is configured (theme, core, images)

### CODE REVIEW

File: core/server/web/site/app.js [hunk 1]
```diff
@@ -44,9 +44,14 @@ module.exports = function setupSiteApp() {
     // you can extend Ghost with a custom redirects file
     // see https://github.com/TryGhost/Ghost/issues/7707
     customRedirects.use(siteApp);
+
     // More redirects
     siteApp.use(adminRedirects());
 
+    // Force SSL if required
+    // must happen BEFORE serving assets and BEFORE routing
```
Comment (ErisDS) [inline core/server/web/site/app.js]: Can we go the extra step of adding a why to this comment?

The old comment had no reason (my fault) and now I don't remember why it was the other way around, so maybe let's not make the same mistake again 😂

**Review by ErisDS — CHANGES_REQUESTED** [review]
Whoohoo tests 🎉  One tiny change plz plz

File: core/server/web/site/app.js (hunk 1 above)
Comment (kirrg001) [reply core/server/web/site/app.js]: 👍

**Comment by runspired** [comment]
Is there a way to tell if our ghost blog has received this fix version?

**Comment by kirrg001** [comment]
@runspired Sure. This fix was shipped in [1.21.2](https://github.com/TryGhost/Ghost/releases/tag/1.21.2). The [Ghost CLI](https://docs.ghost.org/docs/cli-ls) will tell you which version your blog is on.

**Pull request merged**